### PR TITLE
Misha/ob-122_collatz_conjecture

### PR DESCRIPTION
**Pull request by krassovki**

### CODE REVIEW

File: Misha/OB-122_collatz_conjecture/collatz_conjecture.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        iterations += 1
+        print('value[' + str(iterations) + ']: \t' + str(value))
+    return iterations
+
+
+def check_input_data_format(value):
+    if value is None or not re.fullmatch(r'[1-9]{1,}\d{0,}', value) or re.fullmatch(r'[1]', value):
+        raise TypeError('Invalid data format')
+    else:
+        pass
```
Comment (glock3) [inline Misha/OB-122_collatz_conjecture/collatz_conjecture.py]: Why do you need this? What does it do?

File: Misha/OB-122_collatz_conjecture/collatz_conjecture_test.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import unittest
+from collatz_conjecture import collatz_conjecture
+
+
+class CollatzConjectureTest(unittest.TestCase):
```
Comment (glock3) [inline Misha/OB-122_collatz_conjecture/collatz_conjecture_test.py:5]: Please add test cases with valid data and border values

File: Misha/OB-122_collatz_conjecture/collatz_conjecture.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import time
+import re
+
+
+def show_time_of_function_processing(func):
+    def wrapper_with_arguments(arg1):
```
Comment (glock3) [inline Misha/OB-122_collatz_conjecture/collatz_conjecture.py]: Naming
Do you have arg2?

**Review by glock3 — CHANGES_REQUESTED** [review]

File: Misha/OB-122_collatz_conjecture/collatz_conjecture.py (hunk 1 above)
Comment (krassovki) [reply Misha/OB-122_collatz_conjecture/collatz_conjecture.py]: already deleted this

File: Misha/OB-122_collatz_conjecture/collatz_conjecture.py (hunk 3 above)
Comment (krassovki) [reply Misha/OB-122_collatz_conjecture/collatz_conjecture.py]: corrected

File: Misha/OB-122_collatz_conjecture/collatz_conjecture_test.py (hunk 2 above)
Comment (krassovki) [reply Misha/OB-122_collatz_conjecture/collatz_conjecture_test.py:5]: two right values were added before your rewiev - 7 and 12. test_cc_with_right_value_2 and test_cc_with_right_value

File: Misha/OB-122_collatz_conjecture/collatz_conjecture.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import time
```
Comment (glock3) [inline Misha/OB-122_collatz_conjecture/collatz_conjecture.py:1]: Move tasks to Algorithm folder

**Review by glock3 — APPROVED** [review]

**Pull request merged**